### DISCUSSION TITLE
OpenTelemetry warning on SQLAlchemy ORM operation with multiple data types

### DISCUSSION THREAD

**Discussion by gilles-leblanc** [open-telemetry/opentelemetry-python-contrib#1988] [Q&A]
Hi,

I have an application that uses SQLAlchemy ORM to communicate with a database. 
It's instrumented with OpenTelemetry SQLAlchemy Instrumentation.

I'm seeing warnings in the log of the following form:

```
{"message": "Attribute 'db.params' mixes types str and int in attribute value sequence","severity": "WARNING", "logging.googleapis.com/labels": {"python_logger": "opentelemetry.attributes"}, "logging.googleapis.com/trace": "",  "logging.googleapis.com/spanId": "", "logging.googleapis.com/trace_sampled": false, "logging.googleapis.com/sourceLocation": {"line": 86, "file": "/root/.local/lib/python3.10/site-packages/opentelemetry/attributes/__init__.py", "function": "_clean_attribute"}, "httpRequest": {} }
```

Here are the values which triggered the warnings:

`['6e48dfd0-5084-4bf6-b794-f514e290814d', '127.0.0.1', 11111]`

The data types for these values match the data types for the fields in the database as well as the types for the `MatchConnectionDetailsOrm` class itself.

The `MatchConnectionDetailsOrm` is a child property to a parent Orm class `MatchOrm`. 
```
new_match = MatchOrm(
            product_id=product_id,
            server_id=content.server_id,
            connection_details=MatchConnectionDetailsOrm(
                host=content.host,
                port=content.port,
            ),
        )
```

As far as I can tell the values of MatchConnectionDetailsOrm end up as a sequence called params and there is a check inside OpenTelemetry to make sure all values in this sequence should be of the same type. 

As the port is indeed an Integer and  host a string, it's expected that not all values should be of the same type. 

I would expect having a SQL operation with multiple data types not to trigger a warning. 
Is there something that I'm doing wrong?
If not, is there a way to tell prevent logging warnings such as these?

**Comment by gilles-leblanc** [comment]
I've been doing some digging around. 
I'm using Google's Python Spanner SQLAlchemy package.

In it they trace with OpenTelemetry, if installed, and add attributes called db.params which contains the values of the SQL query. These are not guaranteed to be in the same format. 

I'll probably open an issue with the Google project.

**Reply by ocelotl** [reply]
Thanks, @gilles-leblanc can we close this conversation with your latest comment as an answer?
